src/EPR.Payment.Facade/Validations/ResubmissionFees/Producer/ProducerResubmissionFeeRequestDtoValidator.cs

```diff
@@ -14,23 +14,17 @@ public ProducerResubmissionFeeRequestDtoValidator()
                 .Must(RegulatorValidationHelper.IsValidRegulator).WithMessage(ValidationMessages.RegulatorInvalid);
 
             RuleFor(x => x.ResubmissionDate)
-                .NotEmpty().WithMessage(ValidationMessages.ResubmissionDateRequired)
-                .Must(BeInUtc).WithMessage(ValidationMessages.ResubmissionDateMustBeUtc)
-                .Must(BeValidDate).WithMessage(ValidationMessages.ResubmissionDateDefaultInvalid)
-                .LessThanOrEqualTo(DateTime.UtcNow).WithMessage(ValidationMessages.ResubmissionDateInvalid);
+                 .Cascade(CascadeMode.Stop)
+                 .NotEmpty().WithMessage(ValidationMessages.ResubmissionDateRequired)
+                 .Must(BeInUtc).WithMessage(ValidationMessages.ResubmissionDateMustBeUtc)
+                 .LessThanOrEqualTo(DateTime.UtcNow).WithMessage(ValidationMessages.FutureResubmissionDate);
```

Comment:
I noticed that this code is repeated in ProducerResubmissionFeeRequestDtoValidator and ComplianceSchemeResubmissionFeeRequestDtoValidator. I think it would be good to add the ResubmissionDate validation in a common helper validator, like we do for RegulatorValidationHelper.

Please check if this can be done for registration fees also.

Reply:
done